sphinx_construct/documenters/subcon.py

```diff
@@ -60,178 +66,247 @@ def _typename(self, obj):
 
-	def _recuse(self, obj):
+	def _recurse(self, obj, indent : bool = True):
 		if obj in _documented_subcon_instances:
+			name = _documented_subcon_instances[obj]
+			self.append()
+			self.append(f'   See: :py:attr:`{name}`')
 			return
 
-		old_indent = self.indent
-		self.indent = f'{old_indent}   '
+		if indent:
+			old_indent = self.indent
+			self.indent = f'{old_indent}   '
+		if obj.name is not None:
+			name = self.name
+			self.name = f'{name}.{obj.name}'
+
 		if hasattr(obj, 'subcons'):
 			for sc in obj.subcons:
+				self.append()
 				self._subcon_handlers.get(type(sc), self._default_handler)(sc)
 
 		elif hasattr(obj, 'subcon'):
 			self._subcon_handlers.get(type(obj.subcon), self._default_handler)(obj.subcon)
 
-		self.indent = old_indent
-		_documented_subcon_instances.append(obj)
+		if obj.name is not None:
+			self.name = name
+		if indent:
+			self.indent = old_indent
+		_documented_subcon_instances[obj] = self.name
 
 	# -- Type Handlers -- #
 
-	def _enum_handler(self, obj, _ = None):
+	def _enum_handler(self, obj):
 		size   = obj.subcon.sizeof()
-		do_hex = (size & 3) == 0
-		base   = 'x' if do_hex else 'b'
-		size   = (size >> 2) if do_hex else size
+		if self.size_mode == SizeMode.BYTES:
+			base = 'x'
+			size = size * 2
+		else:
+			do_hex = (size & 3) == 0
+			base   = 'x' if do_hex else 'b'
+			size   = (size >> 2) if do_hex else size
 
 		def _val_to_str(value):
 			return f'0{base}{value:0{size}{base}}'
 
-
 		for v, k in obj.ksymapping.items():
-			self.append(f'.. py:attribute:: {str(self.name).replace("::", ".")}.{k}')
-			self.append(f'   :type: {obj.subcon.__class__.__name__}<{size}>')
-			self.append(f'   :value: {_val_to_str(v)}')
 			self.append()
+			self.append(f'.. py:attribute:: {self.name}.{k}')
+			self.append(f'   :type: {self._typename(obj.subcon)}<{size}>')
+			self.append(f'   :value: {_val_to_str(v)}')
 
 		if hasattr(obj, 'docs'):
-			self.append(obj.docs)
 			self.append()
+			self.append(obj.docs)
 
-	def _renamed_handler(self, obj, is_header = False):
-		if is_header:
-			# self.append(f'.. py:attribute:: {obj.name}')
-			# self.append(f'   :module: {self.modname}')
-			self.append(f'   :type: {self._typename(obj.subcon)}')
+	def _flags_enum_handler(self, obj):
+		size   = obj.subcon.sizeof()
+		if self.size_mode == SizeMode.BYTES:
+			base = 'x'
+			size = size * 2
 		else:
-			di = self._documented_instance(obj)
-			if di is None:
-				# self.append(f'.. _{self._mk_tgt_name(obj)}:')
-				self.append()
-				self.append(f'.. py:attribute:: {obj.name}')
-				self.append(f'   :type: {self._typename(obj.subcon)}')
-				self.append( '   :noindex:')
-				self.append()
-				self._recuse(obj)
-			else:
-				self.append(f'`See: "{obj.name}" <{self._mk_tgt_name(obj)}>`_')
-				self.append()
+			do_hex = (size & 3) == 0
+			base   = 'x' if do_hex else 'b'
+			size   = (size >> 2) if do_hex else size
 
+		def _val_to_str(value):
+			return f'0x{value:0{size}{base}}'
```

Comment:
this would be strange if base was `b`, it would produce `0x101010` which is not correct